### PR TITLE
Per 9743 replace vagrant with docker

### PR DESCRIPTION
**Pull request by valentinvsp**
The `docker-compose.yml` file has been modified to include the PHP api as a Docker VM. It also contains the web-app as a separate VM container, as they were previously served together by the Vagrant VM.
For the setup to work, the nginx conf file had to be updated.
Also, the postgres healthcheck config was updated beacuse it was causing an error in the postgres server and was polluting the logs.

Some of the commands that are run in the back-end container are based on the `php:8.3-bookworm` docker image, and this required the presence of certain shell scripts. These have been added in this repo so that they can be included in the image build process. They are subject to the Docker MIT license which is included with the files.

The original Vagrant VM was running multiple processes. Unfortunately Docker does not provide easy access to `systemctl` and the recommended solution is to use a process manager. For this reason **supervisord** was installed and configured.

### CODE REVIEW

**Review by cecilia-donnelly — CHANGES_REQUESTED** [review]
Can you update the README to describe how to use this version?

**Review by liam-lloyd — COMMENTED** [review]
When I run this locally, the API appears to work (though I haven't tested exhaustively), but the daemons don't. This appears to be because [this check](https://github.com/PermanentOrg/back-end/blob/626e4f2a37412d3dfee7f4cb8e1f5af1b3001b29/daemon/index.php#L73) always fails.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-version: "3.7"
```
[the remark this answers is not in the document]
Comment (valentinvsp) [reply docker-compose.yml:1]: afaik docker compose file versioning has been deprecated and you should get some warning in the logs if you still have it

File: docker-compose.yml [hunk 2]
```diff
@@ -11,6 +10,49 @@ services:
     depends_on:
       stela:
         condition: service_healthy
+      web-app:
+        # TODO -> update condition to healty
```
[the remark this answers is not in the document]
Comment (valentinvsp) [reply docker-compose.yml:14]: in an ideal scenario, but not necessary. would need to define a rule for condition healthy

File: docker-compose.yml [hunk 3]
```diff
@@ -11,6 +10,49 @@ services:
...
+        UPLOAD_SERVICE_SENTRY_DSN: ${UPLOAD_SERVICE_SENTRY_DSN}
+        NOTIFICATION_DATABASE_URL: ${NOTIFICATION_DATABASE_URL}
+        NOTIFICATION_FIREBASE_CREDENTIALS: ${NOTIFICATION_FIREBASE_CREDENTIALS}
+    image: back-end
+    env_file:
+      # env file path is relative to docker-compose file
+      # Availability of env vars at build and during run requires separate configs.
+      - ./.env
+    volumes:
+      - ../back-end/api:/data/www/api
```
[the remark this answers is not in the document]
Comment (valentinvsp) [reply docker-compose.yml:45]: hm... that makes sense. would also need to exec into the container and restart the service as well

**Comment by valentinvsp** [comment]
> Can you update the README to describe how to use this version?

done

**Comment by cecilia-donnelly** [comment]
Thanks for getting this to such a good place @valentinvsp ! Closing now with the expectation that @liam-lloyd will pick it up as PER-9787 in October.

**Pull request closed without merging**